src/workloads/execution/DensifyFillCombo.yml

```diff
@@ -0,0 +1,181 @@
+        {^RandomDouble: { min: 0.0, max: 500.0 }},
+        {^RandomDate: {min: "2021-01-01", max: "2023-01-01"}},
+        {^RandomString: {length: 15}}, null], weights: [1, 1, 1, 1, 6]}}
+      numeric: {^RandomInt: {min: 0, max: 2000}}
+    # $densify stages always add a $sort on the field that is being densified, so this phase adds
+    # indexes on the date/number fields so that we aren't performing an in-memory sort.
+    Indexes:
+    - *index
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *db
+  Phases:
+  - *Nop
+  - Repeat: 1
+  - *Nop
+  - Repeat: 1
+  - *Nop
+  - Repeat: 1
+  - *Nop
+
+- Name: DensifyFillWithoutPartition
+  Type: RunCommand
+  Threads: 1
+  Phases:
+  - *Nop
+  - *Nop
+  - Duration: 30 seconds
+    Database: *db
+    Operations:
+    - OperationMetricsName: TestDensifyFillWithoutPartition
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: Collection0
+        pipeline: [
+          $densify: {
+            field: *field,
+            range: {
+              bounds: "full",
+              step: *smallStep
+            }
+          },
+          $fill: {
+            output: {
+              toFillRandomType: {method: "locf"}
+            }
+          }
+        ]
+        cursor: {batchSize: *batchSize}
+        allowDiskUse: true
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+
+- Name: DensifyFillWithPartition
+  Type: RunCommand
+  Threads: 1
+  Phases:
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - Duration: 30 seconds
+    Database: *db
+    Operations:
+    - OperationMetricsName: TestDensifyFillWithPartition
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: Collection0
+        pipeline: [
+          $densify: {
+            field: *field,
+            range: {
+              bounds: "partition",
+              step: *smallStep
+            },
+            partitionByFields: ["partitionKey"]
+          },
+          $fill: {
+            output: {
+              toFillRandomType: {method: "locf"},
+            },
+            partitionByFields: ["partitionKey"]
+          }
+        ]
+        cursor: {batchSize: *batchSize}
+        allowDiskUse: true
+  - *Nop
+  - *Nop
+
+- Name: DensifyFillFullBoundsWithPartition
+  Type: RunCommand
+  Threads: 1
+  Phases:
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - Duration: 30 seconds
+    Database: *db
+    Operations:
+    - OperationMetricsName: TestDensifyFillFullBoundsWithPartition
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: Collection0
+        pipeline: [
+          $densify: {
+            field: *field,
+            range: {
+              bounds: "full",
+              step: *smallStep
+            },
+            partitionByFields: ["partitionKey"]
+          },
+          $fill: {
+            output: {
+              toFillRandomType: {
+                method: "locf"
```

Comment:
I wanted to mention that I couldn't find a way to reasonable combine densify and fill with linear method. We would want to not repeat values in the sort field, but generally the sort field is also where we $densify so we want to not repeat values and also for the values to have gaps (eg one doc contains numeric:5, the next numeric:10). I can see if this is possible in genny, but I thought it best to get this up now

Reply:
Yeah I see the issue. Does the "increment" loader not make this doable?